### ISSUE TITLE
Codes needs cohesive restructuring to make them testable

### ISSUE THREAD

**Issue by ahalim-intel** [intel/sriov-cni#48]
We have done some refactoring recently to so that we can add structured unit tests. However, there are still some more refactoring need to be done. As we can see some challenges with mocking netlink functions and some side-effects of doing so. see #42  

Here's what I suggest we can do to improve the code-base:

We are using netlink library in following pkgs/files: 

- sriov/sriov.go
- pkg/config.go

Here is what I suggest we should do in upcoming days:

1.	Move sriov/main.go into a new cmd/main.go directory – add integration test for this file
2.	Move sriov/sriov.go into new package named sriov into pkg/sriov/sriov.go
3.	Define an Interface as NetlinkManager pkg/sriov/sriov.go with all of the netlink functions that we use in the pkg/sriov/sriov.go –so we can mock them at once & only place in its unit-test file
4.	Move AssignFreeVF() function from pkg/config.go into pkg pkg/sriov/sriov.go so the netlink library only to be imported there. We have to mock it for unit tests only once in here. Netlink no longer needed for config.go

**Comment by ahalim-intel** [comment]
@krsacme 
Just to let you know, I was informed that @pmossakx is taking care of this issue as he is already working on implementing additional tests.

**Comment by killianmuldoon** [comment]
@ahalim-intel I think this has been done at this stage, right?

**Comment by ahalim-intel** [comment]
@killianmuldoon yes, It's done
